src/docs/commands_keybindings.md

```diff
@@ -1,122 +1,142 @@
 ---
-title: Commands and Keybindings
+title: Commands/Menus/Keybindings
 ---
 
-# Commands and Keybindings
+# Commands, Menus and Keybindings
 
-Theia can be extended in a number of different ways. Commands allow packages to contribute unique commands that can then be called by other packages. It's also possible to add keybindings and contexts to these commands, so that they can only be called under certain conditions (window focus, current selection etc.).
+Commands are runnable actions defined by an ID and the function to be executed (plus some optional parameters like a name or an icon). Commands can be triggered via the command pallete, they can be bound to keybindings or menu items and they can be called programmatically. The action that commands trigger can be context sensitive, so that they can only be called under certain conditions (window focus, current selection etc.).
 
-## Adding commands to Theia command registry
+The following sections provide details about how to contribute commands, keybindings and menu items. The sections will describe how to connect the different contributions and how to use the corresponding services for managing these items.
 
-To contribute commands to the command registry, modules must implement the `CommandContribution` class, i.e
+If you are not yet familiar with contribution points in Theia or the use of dependency injection, please consider this guide on [Services and Contributions](https://theia-ide.org/docs/services_and_contributions/).
 
-**java-commands.ts**
+All the following code examples are from the [Theia extension generator](https://github.com/eclipse-theia/generator-theia-extension). You can get the same code set-up by installing the generator, selecting the “Hello World” example (see here) and choosing “helloworld” as the name. 
+
+## Contributing commands
+
+By contributing a command, you can add a custom action to Theia. A command can be triggered by the user via the quick access bar, a keybinding or a menu entry. It can also be called programmatically.
```

Comment:
```suggestion
By contributing a command, you can add a custom action to Theia. A command can be triggered by the user via the command palette, a keybinding or a menu entry. It can also be called programmatically.
```